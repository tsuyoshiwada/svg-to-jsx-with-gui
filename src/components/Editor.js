/* eslint-disable */
import React, { Component } from 'react';
import styled from 'styled-components';
import CodeMirror from 'react-codemirror';

const EditorHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
  z-index: 10;
  height: 40px;
  padding: 0 20px;
  background: #31364f;
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.22);
  color: #fff;
`;

const EditorTitle = styled.div`
  flex: 0 1 auto;
  font-size: 1.14rem;
`;

const EditorWrapper = styled.div`
  position: relative;
  height: calc(100vh - 60px);
  background: #353954;
  font-size: 14px;
  line-height: 1.6;

  & .CodeMirror {
    position: absolute;
    top: 50px;
    right: 0;
    left: 0;
    height: calc(100% - 50px);
  }

  & .CodeMirror,
  & .CodeMirror-gutters {
    background: #353954 !important;
  }

  & .CodeMirror-linenumber {
    color: #4c5275 !important;
  }

  & .CodeMirror-scrollbar-filler {
    background: transparent;
  }
`;

export default class Editor extends Component {
  static defaultProps = {
    readOnly: false,
    options: {
      mode: 'html',
      theme: 'material',
      lineNumbers: true,
    },
  };

  componentWillReceiveProps(nextProps) {
    const { value: _value } = this.props;
    const { value, readOnly } = nextProps;

    if (readOnly && value !== _value && this.editor) {
      this.editor.getCodeMirror().doc.setValue(value);
    }
  }

  handleRef = (editor) => {
    this.editor = editor;
  };

  render() {
    const {
      title,
      mode,
      readOnly,
      options,
      ...rest
    } = this.props;

    return (
      <EditorWrapper>
        <EditorHeader>
          <EditorTitle>
            {title}
          </EditorTitle>
        </EditorHeader>

        <CodeMirror
          {...rest}
          ref={this.handleRef}
          options={{
            ...options,
            mode,
            readOnly,
          }}
        />
      </EditorWrapper>
    );
  }
}
