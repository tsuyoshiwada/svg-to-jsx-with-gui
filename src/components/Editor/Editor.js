import React, { Component } from 'react';
import styled from 'styled-components';
import AceEditor from 'react-ace';
import 'brace/mode/html'; // eslint-disable-line import/no-extraneous-dependencies
import 'brace/mode/jsx'; // eslint-disable-line import/no-extraneous-dependencies
import 'brace/theme/monokai'; // eslint-disable-line import/no-extraneous-dependencies
import { palette, darken, brighten } from '../../styles';

const EditorHeader = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
  z-index: 10;
  height: 40px;
  padding: 0 20px;
  background: ${palette.secondaryDark};
  box-shadow: 0 1px 2px rgba(0, 0, 0, 0.22);
  color: #fff;
`;

const EditorTitle = styled.div`
  flex: 0 1 auto;
  font-size: 1.14rem;
`;

const EditorWrapper = styled.div`
  position: relative;
  height: calc(100vh - 60px);
  background: ${palette.secondary};
  font-size: 14px;
  line-height: 1.6;

  & .ace_editor {
    position: absolute;
    top: 40px;
    right: 0;
    left: 0;
    width: 100% !important;
    height: calc(100% - 40px) !important;
    color: #ccc !important;
  }

  & .ace_editor,
  & .ace_gutter {
    background: ${palette.secondary} !important;
  }

  & .ace_marker-layer .ace_selection {
    background: ${brighten(palette.secondary, 0.4)} !important;
  }

  & .ace_gutter {
    color: ${palette.secondaryLight} !important;
  }

  & .ace_gutter-active-line,
  & .ace_marker-layer .ace_active-line {
    background: ${darken(palette.secondary, 0.1)} !important;
  }

  & .ace_scroller.ace_scroll-left {
    transition: box-shadow 120ms ease-out;

    &.ace_scroll-left {
      box-shadow: 17px 0 16px -16px rgba(0, 0, 0, 0.2) inset !important;
    }
  }

  & .ace_entity,
  & .ace_keyword,
  & .ace_meta.ace_tag,
  & .ace_storage {
    color: #d99 !important;
  }

  & .ace_entity.ace_name.ace_function,
  & .ace_entity.ace_other,
  & .ace_entity.ace_other.ace_attribute-name,
  & .ace_variable {
    color: #71ddf1 !important;
  }

  & .ace_string {
    color: #fff8b7 !important;
  }

  & .ace_storage.ace_type,
  & .ace_support.ace_class,
  & .ace_support.ace_type {
    color: #5092e3;
    font-style: normal !important;
  }

  & .ace_comment {
    color: #818495 !important;
  }
`;

const EditorButtonList = styled.div`
  margin-right: -20px;
  text-align: right;
`;

export default class Editor extends Component {
  static defaultProps = {
    showGutter: false,
    useSoftTabs: false,
  };

  handleRef = (editor) => {
    this.editor = editor;
  };

  render() {
    const {
      title,
      buttons,
      useSoftTabs,
      ...rest
    } = this.props;

    return (
      <EditorWrapper>
        <EditorHeader>
          <EditorTitle>
            {title}
          </EditorTitle>
          {buttons &&
            <EditorButtonList>
              {buttons.map((button, index) => (
                React.cloneElement(button, {
                  ...button.props,
                  key: index, // eslint-disable-line react/no-array-index-key
                })
              ))}
            </EditorButtonList>
          }
        </EditorHeader>

        <AceEditor
          editorProps={{
            $blockScrolling: true,
          }}
          {...rest}
          showPrintMargin={false}
          setOptions={{
            displayIndentGuides: true,
            useSoftTabs,
          }}
          theme="monokai"
          ref={this.handleRef}
        />
      </EditorWrapper>
    );
  }
}
