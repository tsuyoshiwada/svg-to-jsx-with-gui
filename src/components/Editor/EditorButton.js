import React, { Component } from 'react';
import styled from 'styled-components';
import { palette } from '../../styles';
import Button from '../internal/Button';


const EditorButtonWrapper = Button.extend`
  padding: 0 20px;
  height: 40px;
  border-left: 1px solid rgba(255, 255, 255, 0.1);
  background: ${palette.secondaryDark};
  color: #fff;
  font-size: 0.75rem;

  &:hover {
    background-color: ${palette.secondaryLight};
  }

  &:active {
    background-color: #393f5d;
  }
`;

const IconWrapper = styled.span`
  display: inline-block;
  width: 1em;
  height: 1em;
  margin: 0 0.5em 0 0;

  & svg {
    width: 1em;
    height: 1em;
    fill: #fff;
  }
`;


export default class EditorButton extends Component {
  handleClick = (e) => {
    e.preventDefault();
    if (typeof this.props.onClick === 'function') {
      this.props.onClick(e);
    }
  };

  render() {
    const {
      children,
      icon,
      onClick,
      ...rest
    } = this.props;

    return (
      <EditorButtonWrapper
        {...rest}
        onClick={this.handleClick}
      >
        {icon &&
          <IconWrapper>{icon}</IconWrapper>
        }
        {children}
      </EditorButtonWrapper>
    );
  }
}
