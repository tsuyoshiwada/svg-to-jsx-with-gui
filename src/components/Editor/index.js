// @flow
export { default as Editor } from './Editor';
export { default as EditorButton } from './EditorButton';
export { default as EditorCopyButton } from './EditorCopyButton';
export { default as EditorDownloadButton } from './EditorDownloadButton';
export { default as EditorSplit } from './EditorSplit';
