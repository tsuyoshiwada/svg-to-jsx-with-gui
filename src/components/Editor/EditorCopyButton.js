import React, { Component } from 'react';
import withCopy from '../hoc/withCopy';
import EditorButton from './EditorButton';
import { Copy } from '../icons/';

// eslint-disable-next-line react/prefer-stateless-function
class EditorCopyButton extends Component {
  render() {
    return (
      <EditorButton
        {...this.props}
        icon={<Copy />}
      />
    );
  }
}

export default withCopy(EditorCopyButton);
