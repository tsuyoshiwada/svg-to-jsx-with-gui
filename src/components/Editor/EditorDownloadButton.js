import React from 'react';
import EditorButton from './EditorButton';
import { Download } from '../icons/';

const EditorDownloadButton = props => (
  <EditorButton
    icon={<Download />}
    {...props}
  />
);

export default EditorDownloadButton;
