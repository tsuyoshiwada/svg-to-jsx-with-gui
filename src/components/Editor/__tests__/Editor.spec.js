import React from 'react';
import { shallow } from 'enzyme';
import Editor from '../Editor';


describe('components#<Editor />', () => {
  test('Should be render', () => {
    const wrapper = shallow(<Editor />);
    expect(wrapper).toMatchSnapshot();
  });
});
