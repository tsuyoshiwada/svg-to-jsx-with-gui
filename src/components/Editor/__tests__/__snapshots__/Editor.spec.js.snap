// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components#<Editor /> Should be render 1`] = `
<styled.div>
  <styled.div>
    <styled.div />
  </styled.div>
  <ReactAce
    cursorStart={1}
    editorProps={
      Object {
        "$blockScrolling": true,
      }
    }
    enableBasicAutocompletion={false}
    enableLiveAutocompletion={false}
    focus={false}
    fontSize={12}
    height="500px"
    highlightActiveLine={true}
    maxLines={null}
    minLines={null}
    mode=""
    name="brace-editor"
    onChange={null}
    onLoad={null}
    onPaste={null}
    onScroll={null}
    readOnly={false}
    scrollMargin={
      Array [
        0,
        0,
        0,
        0,
      ]
    }
    setOptions={
      Object {
        "displayIndentGuides": true,
        "useSoftTabs": false,
      }
    }
    showGutter={false}
    showPrintMargin={false}
    style={Object {}}
    tabSize={4}
    theme="monokai"
    value=""
    width="500px"
    wrapEnabled={false}
  />
</styled.div>
`;
